packages/lib/gpc/src/gpc.ts

```diff
@@ -0,0 +1,172 @@
+import {
+  ProtoPODGPC,
+  ProtoPODGPCCircuitDesc,
+  gpcArtifactPaths
+} from "@pcd/gpcircuits";
+import { Groth16Proof } from "snarkjs";
+import {
+  checkCircuitParameters,
+  checkProofArgs,
+  checkProofConfig,
+  checkVerifyArgs
+} from "./gpcChecks";
+import {
+  compileProofConfig,
+  compileVerifyConfig,
+  makeRevealedClaims
+} from "./gpcCompile";
+import {
+  GPCBoundConfig,
+  GPCIdentifier,
+  GPCProofConfig,
+  GPCProofInputs,
+  GPCRevealedClaims
+} from "./gpcTypes";
+import { canonicalizeConfig, makeCircuitIdentifier } from "./gpcUtil";
+
+// TODO(POD-P1): Consider whether all the separate checking and canonicalizaion
+// are valuable, or if some can be eliminated in favor of simply letting
+// compilation crash on bad inputs.
+
+/**
+ * Checks, binds, and canonicalizes a GPCProofConfig so it can be reused
+ * for multiple proofs.  See {@link GPCBoundConfig} for more details.
+ *
+ * Note that this function cannot verify anything about future inputs.  In
+ * particular the max POD size supported by an auto-selected circuit might
+ * not be sufficient for all inputs.  If you need a larger size, you should pick
+ * your circuit explicitly using the circuitIdentifier argument.
```

Comment:
I'll add a link to the array in gpcircuits, though realistically it won't be very useful until we have a more stable set of circuits, at which point we should have public docs of the supported flavors in a readme.  Meanwhile I think most
people doing this will either know the code, or can run a test case with a max-sized POD to get the answer they need.